### ISSUE TITLE
Using GINKGO_FOCUS and/or GINKGO_SKIP overwrites kubetest variables

### ISSUE THREAD

**Issue by chrischdi** [kubernetes-sigs/cluster-api#9634]
### What steps did you take and what happened?

* Create a new yaml file in `test/e2e/data/kubetest` which defines ginkgo.skip and/or ginkgo.focus
  * to e.g. run the slow tests with vars from: https://github.com/kubernetes/test-infra/blob/92927d1abec1c3c78e396356e1bbaf19f9e2b699/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L917
* Run conformance tests with this file:
  * Using `GINKGO_FOCUS="\\[Conformance\\]"` to only run kubetest
  * Using `KUBETEST_CONFIGURATION="./data/kubetest/slow.yaml"

* The CAPI Conformance test started and kubetest e2e tests started and did run.
* The kubetest / upstream kubernetes test suite did run with ginkgo focus of `\\[Conformance\\]` instead of what was defined in `test/e2e/data/kubetest/slow.yaml`

### What did you expect to happen?

E2E tests to run with the `ginkgo.focus` variable set at `test/e2e/data/kubetest/slow.yaml`

### Cluster API version

main

### Anything else you would like to add?

`GINKGO_*` variables leak down into the function `test/framework/kubetest/run.go.Run` which executes the upstream e2e test suite.

* I wanted to create multiple jobs in prow for CAPV.
* Each of the jobs should have referred a different config file via the `KUBETEST_CONFIGURATION` variable
* Each of the jobs run with `GINKGO_FOCUS="\\[Conformance\\]"` to run the conformance tests, but with the above kubetest configuration

This is currently not possible because the `GINKGO_FOCUS` and also other GINKGO variables (e.g. `GINKGO_SKIP` leak down into the ginkgo run done for kubetest.

A quick workaround was to change the Makefile of CAPI (or on my case CAPV) to use `CAPI_GINKGO_FOCUS` instead of `GINKGO_FOCUS` inside the makefile.

### Label(s) to be applied

/kind bug

/area e2e-testing

**Comment by killianmuldoon** [comment]
/triage accepted

This is an annoying one :neutral_face:. Do the variables get passed into the kubetest image itself? i.e. at `https://github.com/kubernetes-sigs/cluster-api/blob/6fdd6a3248113e9d313d253691bcba50d452bed4/test/framework/kubetest/run.go#L190`?

If so maybe we can have some handling in our Kubetest runner to take this into account, and have something like `KUBETEST_GINKGO_SKIP` which would simplify the changes we'd need to make for CAPI's test infra and run configurations.

**Comment by chrischdi** [comment]
> /triage accepted
> 
> This is an annoying one 😐. Do the variables get passed into the kubetest image itself? i.e. at `https://github.com/kubernetes-sigs/cluster-api/blob/6fdd6a3248113e9d313d253691bcba50d452bed4/test/framework/kubetest/run.go#L190`?
> 
> If so maybe we can have some handling in our Kubetest runner to take this into account, and have something like `KUBETEST_GINKGO_SKIP` which would simplify the changes we'd need to make for CAPI's test infra and run configurations.

jep, having a `KUBETEST_GINKGO_SKIP` which gets propagated into the kubetest env as `GINKGO_SKIP` (instead of the real `GINKGO_SKIP`) may work too.

**Comment by chrischdi** [comment]
To confirm: `GINKGO_FOCUS` leaks down as `E2E_FOCUS` into the conformance pod:

```
❯ docker inspect eda6fec95b1b | grep Env -A 10
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "E2E_FOCUS=\\[Conformance\\]",
```

**Comment by chrischdi** [comment]
Oh damn, I was wrong, looks like I did take a look at the wrong junit xml file, interpreting the properties there.

TLDR: this works. If one wants to compare the properties in a junit xml file: use `_artifacts/junit.kubetest.01.xml` instead of  `_artifacts/junit.e2e_suite.1.xml`.

GINKGO_SKIP and GINKGO_FOCUS do net propagated down into the ginkgo running for conformance.

/close
